sys/include/net/gnrc/netdev2.h

```diff
@@ -81,8 +96,59 @@ typedef struct gnrc_netdev2 {
      * @brief PID of this adapter for netapi messages
      */
     kernel_pid_t pid;
+
+#ifdef MODULE_GNRC_MAC
+    /**
+     * @brief general information for the MAC protocol
+     */
+    uint16_t mac_info;
+
+    /**
+     * @brief device's l2 address
+     */
+    uint8_t  l2_addr[GNRC_NETIF_HDR_L2ADDR_MAX_LEN];
```

Comment:
@miri64 , updated according to your last comment, moved `l2_addr_t` into `gnrc_netdev2_t` without creating a new type. Is this OK for you? :-)